### PR TITLE
Properly isolate tests to enable concurrent execution.

### PR DESCRIPTION
**Pull request by SamBarker**
I've been unable to reproduce the flaky test failures locally using:

```bash
#!/usr/bin/env bash

exit=0
while [[ ${exit} == 0 ]]; do
  mvn -B verify
  exit=$?
done
```

However these commits should make the test better able to run in parallel.

### CODE REVIEW

**Comment by k-wall** [comment]
@SamBarker I notice before my change, when I ran the change locally, the surefire tests always ran in fork mode, one test class per fork.  This was masking a whole heap of sins.  On CI, the tests were running in a single VM, in parallel.

**Comment by SamBarker** [comment]
Hum wonder why it's different given the Pom is consistent. I can try with explicitly disabling the forking locally.

**Comment by k-wall** [comment]
How does naming the appender help?  If tests run in parallel surely one test will hear the other tests output as the output will go to all appenders.   It did occur to me that injecting a logger into the class under test might be a better way to achieve isolation.

**Comment by k-wall** [comment]
> Hum wonder why it's different given the Pom is consistent. I can try with explicitly disabling the forking locally.

Don't know. I fell down a rabbit hole trying to work this out trying to explain the difference in behaviour.  Answers on SO were not helpful.


> 
> On the or I was getting regular failures without my fix and then got three green builds in a row.

**Comment by SamBarker** [comment]
Running main with the script in the PR description and forcing as much concurrency as possible I can re-reproduce the build failure locally (less 90s with the script). 

Switching to my branch and applying the maximum concurrency again the PR passes build after build (> 30mins without failure). Given turning up the concurrency level seems to speed things up locally I've included it in this PR.

Pom changes for maximum concurrency:
```
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision 3bed3a19f58216296eaab4431c5b42f248d45e52)
+++ b/pom.xml	(date 1657501160120)
@@ -143,8 +143,13 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
                     <redirectTestOutputToFile>true</redirectTestOutputToFile>
-                    <useUnlimitedThreads>false</useUnlimitedThreads>
+                    <useUnlimitedThreads>true</useUnlimitedThreads>
+                    <parallel>all</parallel>
+                    <useUnlimitedThreads>true</useUnlimitedThreads>
                     <forkCount>1</forkCount>
+                    <reuseForks>true</reuseForks>
+                    <forkedProcessTimeoutInSeconds>90</forkedProcessTimeoutInSeconds>
+                    <forkedProcessExitTimeoutInSeconds>90</forkedProcessExitTimeoutInSeconds>
                 </configuration>
             </plugin>
             <plugin>
```

File: pom.xml [hunk 1]
```diff
@@ -162,6 +166,7 @@
                             <goal>integration-test</goal>
                             <goal>verify</goal>
                         </goals>
+                        <phase>install</phase>
```
Comment (showuon) [inline pom.xml:169]: Is this necessary?

File: src/main/java/io/bf2/kafka/authorizer/AuditLoggingController.java [hunk 2]
```diff
@@ -61,6 +61,22 @@ public class AuditLoggingController implements Configurable, Closeable {
...
+     * Primary constructor
+     */
+    public AuditLoggingController() {
+        this("AuditEvents");
+    }
+
+    /**
+     * primarily for tests to increase isolation.
+     * Identifies which logger to direct audit events too.
+     * @param auditLoggerKey the name for
```
Comment (showuon) [inline src/main/java/io/bf2/kafka/authorizer/AuditLoggingController.java:74]: 1. `Identifies which logger to direct audit events too.` -> Should we remove the `too` in the end? I don't know what it means here
2. @param auditLoggerKey the name for [WHAT?]

File: src/test/java/io/bf2/kafka/authorizer/VerifiableAppenderExtension.java [hunk 3]
```diff
@@ -13,48 +13,66 @@
...
+            return extensionContext.getStore(namespace).get(LOG_EVENTS_STORE_KEY, List.class);
+        }
+        if (parameterContext.findAnnotation(LoggerName.class).isPresent()) {
+            return getLoggerName(extensionContext);
+        }
+        return null;
+    }
+
+    private String getLoggerName(ExtensionContext extensionContext) {
+        return extensionContext.getStore(namespace).getOrComputeIfAbsent("LoggerName", key -> "AuditEvents." + Instant.now(), String.class);
```
Comment (showuon) [inline src/test/java/io/bf2/kafka/authorizer/VerifiableAppenderExtension.java:75]: nit: should we add a variable for `LoggerName` like `LOG_EVENTS_STORE_KEY` did?

File: src/test/java/io/bf2/kafka/authorizer/VerifiableAppenderExtension.java [hunk 4]
```diff
@@ -13,48 +13,66 @@
...
     private final ExtensionContext.Namespace namespace;
 
     public VerifiableAppenderExtension() {
         namespace = ExtensionContext.Namespace.create(VerifiableAppenderExtension.class);
+        appenderName = "VerifiableAppender-" + Instant.now();
     }
 
     @Override
     public void afterEach(ExtensionContext extensionContext) {
         extensionContext.getStore(namespace).remove(LOG_EVENTS_STORE_KEY, List.class);
```
Comment (showuon) [inline src/test/java/io/bf2/kafka/authorizer/VerifiableAppenderExtension.java:41]: Should we also remove `LoggerName` from store namespace after each test?

**Review by showuon — COMMENTED** [review]
Thanks for the improvement and tests. Left some comments.

File: pom.xml (hunk 1 above)
Comment (SamBarker) [reply pom.xml:169]: Oops. No that wasn't meant to be committed.

File: src/main/java/io/bf2/kafka/authorizer/AuditLoggingController.java (hunk 2 above)
Comment (SamBarker) [reply src/main/java/io/bf2/kafka/authorizer/AuditLoggingController.java:74]: 1. Should be `to` not `too`. Trying to convey the direction of travel of the audit events.
2. Good catch.

File: src/test/java/io/bf2/kafka/authorizer/VerifiableAppenderExtension.java (hunk 4 above)
Comment (SamBarker) [reply src/test/java/io/bf2/kafka/authorizer/VerifiableAppenderExtension.java:41]: Yes.

**Review by showuon — APPROVED** [review]

**Comment by showuon** [comment]
Sorry, pom.xml merge conflict happened after my change. Please help update it.

**Pull request merged**